### PR TITLE
Html css upgrade

### PR DESCRIPTION
**Pull request by ihsan314**
Thanks to Michaela I came across [these templates](https://html5up.net). I ignored CSS all this while since I didn't want to spend time laying things out, but now I have no excuse to freshen up my website. So I integrated [this template](https://html5up.net/big-picture) and in the process caught some bugs in my previous HTML/JavaScript code while trying the website out.

As for formatting, I guess I can do it, but I'll wait for preliminary approval first before reformatting.

Here are screenshots of each of the three views:

- [First view](https://screenshot.googleplex.com/adFjQrnzk2x)
- [Second view](https://screenshot.googleplex.com/Fr9JqpAOZQu)
- [Third view](https://screenshot.googleplex.com/67WskeSXSzk)

### CODE REVIEW

**Review by mmurray22 — CHANGES_REQUESTED** [review]
This is great! Can you include a screenshot of what your website looks like now?

Also it seems that the links you are using don't correctly navigate to the desired page FYI :)

**Comment by ihsan314** [comment]
Ok I now updated the PR description with screenshots and adjusted the links in the PR.

File: portfolio/src/main/webapp/assets/css/images/poptrox-nav.svg [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="96px" height="64px" viewBox="0 0 96 64" zoomAndPan="disable">
+
```
Comment (mmurray22) [inline portfolio/src/main/webapp/assets/css/images/poptrox-nav.svg:2]: not sure if this was auto added or not (if it was automatic, don't worry about it) but these spaces can probably be removed

File: portfolio/src/main/webapp/assets/js/main.js [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+			setTimeout(function() {
+				$body.removeClass('is-preload');
+			}, 100);
+		});
+
+	// Touch mode.
+		if (browser.mobile)
+			$body.addClass('is-touch');
+		else {
+
```
Comment (mmurray22) [inline portfolio/src/main/webapp/assets/js/main.js:35]: if this was auto-added don't worry about it, but all this line spacing is too much

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -2,32 +2,83 @@
...
+	<ul>
+	  <li><a href="#intro">Intro</a></li>
+	  <li><a href="#looks">What I Look Like</a></li>
+	  <li><a href="#more">Learn More</a></li>
+	</ul>
+      </nav>
+    </header>
+    <section id="intro" class="main style1 dark fullscreen">
+      <div class="content">
+	<header>
```
Comment (mmurray22) [inline portfolio/src/main/webapp/index.html:25]: indentation is a bit off but dw about it for now :)

**Review by mmurray22 — APPROVED** [review]
Overall looks interesting! We should definitely look into using a template like this for our website. Not going to hang on any styling issues, but just left a few comments for future consideration :)

File: portfolio/src/main/webapp/index.html (hunk 3 above)
Comment (ihsan314) [reply portfolio/src/main/webapp/index.html:25]: Oh ok this is the only indentation mistake that can be attributed to me, the others are auto-added.

Comment (ihsan314) [reply portfolio/src/main/webapp/index.html:25]: Ah vim autoreplaced 8 spaces with a tab I think, which might explain the inconsistent indentation here. I'll look into this later.

File: portfolio/pom.xml [hunk 4]
```diff
@@ -34,11 +34,10 @@
         <artifactId>appengine-maven-plugin</artifactId>
         <version>2.2.0</version>
         <configuration>
-          <!-- TODO: set project ID. -->
-          <deploy.projectId>YOUR_PROJECT_ID_HERE</deploy.projectId>
+          <deploy.projectId>iolawale-step-2020</deploy.projectId>
```
Comment (ccondit) [inline portfolio/pom.xml]: I would make this change in a separate PR, since it's not related to the HTML / CSS upgrade

File: portfolio/src/main/webapp/index.html [hunk 5]
```diff
@@ -2,32 +2,83 @@
 <html>
   <head>
     <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no" />
```
Comment (ccondit) [inline portfolio/src/main/webapp/index.html]: [Style nit](https://google.github.io/styleguide/htmlcssguide.html#Document_Type) - don't close void elements.

Comment (ccondit) [reply portfolio/src/main/webapp/index.html]: Also with the `link` element below

**Review by ccondit — CHANGES_REQUESTED** [review]
This generally LGTM - the formatting looks way off, but I know that's from the imported css / javascript. We can follow up and get some best practices for how to leverage templates like this in the capstone.

I also have a concern about the minified vs non minified js - but we can follow up offline about it.

File: portfolio/pom.xml (hunk 4 above)
Comment (ihsan314) [reply portfolio/pom.xml]: Ah ok, will move that to another PR.

File: portfolio/src/main/webapp/index.html (hunk 5 above)
Comment (ihsan314) [reply portfolio/src/main/webapp/index.html]: Oh woops that was also from the template, will fix

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -33,7 +33,7 @@ function addRandomGreeting() {
 function addRandomFact() {
   // Keep track of how many times the function has been called, to determine
   // when to reveal the quiz
-  if (typeof addRandomFact.numTimesCalled === undefined) {
+  if (typeof addRandomFact.numTimesCalled === 'undefined') {
```
Comment (ccondit) [inline portfolio/src/main/webapp/script.js:36]: nit - i don't think this should need quotes, does it?

**Review by ccondit — APPROVED** [review]
We can follow up about how we want to handle formatting something like this later. I'm good for you to merge this into your portfolio.

File: portfolio/src/main/webapp/script.js (hunk 6 above)
Comment (ihsan314) [reply portfolio/src/main/webapp/script.js:36]: I tried without quotes and this condition was not executed. I had to add it back. It's possible that if I directly compared `addRandomFact.numTimesCalled` to `undefined` I wouldn't need quotes, but I didn't try that.

**Comment by kaveh096** [comment]
LGTM
Sorry for delay. I've made a pass. But there is no way i can review all these changes in reasonable time. And I don't think it's review-able given it's copied from some other place. I'me not sure if it's 100% copy or you've made changes yourself or added other things in addition to the template in this PR? Anyway, since we're moving away from weekly assignment repos, let's not worry about it.
